GCRCatalogs/alphaq.py

```diff
@@ -357,7 +363,15 @@ def _get_native_quantity_info_dict(self, quantity, default=None):
             if quantity_key not in fh:
                 return default
             modifier = lambda k, v: None if k == 'description' and v == b'None given' else v.decode()
-            return {k: modifier(k, v) for k, v in fh[quantity_key].attrs.items()}
+            return_qty = {k: modifier(k, v) for k, v in fh[quantity_key].attrs.items()}
+            # a hot fix of the units of native halo mass (hostHaloMass) and x for v3+
+            if self.catalog_version >= StrictVersion('3.0'):
+                if quantity == 'hostHaloMass':
+                    return_qty = {'units': 'Msun/h'}
+            else:
+                if quantity == 'x' or quantity == 'y' or quantity == 'z':
+                    return_qty = {'units': 'comoving Mpc'}
```

Comment:
Lines 368 to 373: (i) make each hotfix indenepent (easier future maintenance) and (ii) don't overwrite the full `return_qty`, just update the unit. 
```python
            if self.catalog_version >= StrictVersion('3.0') and quantity == 'hostHaloMass':
               return_qty['units'] = 'Msun/h'
            if self.catalog_version < StrictVersion('3.0') and quantity in set('xyz'):
               return_qty['units'] = 'comoving Mpc'
```